staging/src/k8s.io/client-go/tools/cache/reflector.go

```diff
@@ -297,6 +298,18 @@ func NewReflectorWithOptions(lw ListerWatcher, expectedType interface{}, store R
 	}
 
 	r.useWatchList = clientfeatures.FeatureGates().Enabled(clientfeatures.WatchListClient)
+	if r.useWatchList {
+		if watchlist.DoesClientNotSupportWatchListSemantics(lw) {
+			// Using klog.TODO() here because switching to a caller-provided contextual logger
```

Comment:
@serathius maybe you can comment, i've decided to use `klog.TODO` because switching to the contextual logger would require changing the API and all call sites (which can be done independently of this work).